src/constbuffer_array_splitter.c

```diff
@@ -208,3 +208,150 @@ CONSTBUFFER_ARRAY_HANDLE constbuffer_array_splitter_split(CONSTBUFFER_ARRAY_HAND
 
     return result;
 }
+
+CONSTBUFFER_ARRAY_HANDLE* constbuffer_array_splitter_split_to_array_of_array(CONSTBUFFER_ARRAY_HANDLE buffers, uint32_t max_buffer_size)
+{
+    CONSTBUFFER_ARRAY_HANDLE* result;
+
+    if (buffers == NULL || max_buffer_size == 0)
+    {
+        LogError("Invalid args : CONSTBUFFER_ARRAY_HANDLE buffers = %p, size_t max_buffer_size = %" PRIu32,
+            buffers, max_buffer_size);
+        result = NULL;
+    }
+    else
+    {
+        uint32_t buffer_count;
+        (void)constbuffer_array_get_buffer_count(buffers, &buffer_count);
+
+        if (buffer_count == 0)
+        {
+            result[0] = constbuffer_array_create_empty();
```

Comment:
> result[0] [](http://example.com/codeflow?start=11&length=10)

this is accessing uninitialized memory

Reply:
need to malloc (e.g. `malloc_2` with a count of 1 and size `sizeof(CONSTBUFFER_ARRAY_HANDLE)`)

Reply:
Related to this, I think Dan suggested making the result `TARRAY(CONSTBUFFER_ARRAY_HANDLE)`.

TARRAY does two things:

1. The array is ref-counted
2. The array can grow as needed by calling "ensure capacity".

The second point may help for the normal case below where we split to multiple constbuffer arrays.

Alternatively you would need to iterate over everything once to figure out how many CONSTBUFFER_ARRAY_HANDLE's you need (how many splits), and then allocate the array, then create the array entries.